plugins/scaffolder/src/components/fields/EntityPicker/EntityPicker.tsx

```diff
@@ -81,38 +79,40 @@ export const EntityPicker = (props: EntityPickerProps) => {
   const getLabel = useCallback(
     (ref: string) => {
       try {
-        return humanizeEntityRef(
-          parseEntityRef(ref, { defaultKind, defaultNamespace }),
-          {
-            defaultKind,
-            defaultNamespace,
-          },
-        );
+        return entityPresentationApi.forEntity(ref, {
```

Comment:
So here's the problem; the "ref" here isn't actually a ref, it's some form of a shorthand of a ref. So it's unfortunate that it was named "ref" in the first place if that's the case.

Seems you should do `stringifyEntityRef(parseEntityRef(ref, { defaultKind, defaultNamespace }))` and pass that to `forEntity`.